### DISCUSSION TITLE
Mobx observe any change in the graph

### DISCUSSION THREAD

**Discussion by jtwigg** [mobxjs/mobx#3454] [Q&A]
Mobx is great at only observing changes for things I care about.

But for serialization, whenever anything in the graph changes, I want to queue a LocalStorage each time (albeit behind a _.throttle to make it performant)

Is there a function to `observeAll(store)` just like you would get with `reduxStore.subscribe()` ?

**Comment by kubk** [comment]
Regarding observing any change in the graph - you may try using `toJS` inside `autorun`: https://codesandbox.io/s/youthful-morse-wfvfch

Regarding the persisting a store to localStorage - I'd suggest to use `mobx-persist-store`: https://github.com/quarrant/mobx-persist-store

I am using it in production. It also allows to specify the [delay](https://github.com/quarrant/mobx-persist-store#storageoptions--reactionoptions):

> delay (Number) - Allows you to set a delay option to limit the amount of times the write function is called. No delay by default.
For example if you have a 200 millisecond delay and two changes happen within the delay time then the write function is only called once. If you have no delay then the write function would be called twice.

**Comment by mweststrate** [comment]
See also `deepObserve` in mobx-utils.
